packages/nodes-base/nodes/Git/Git.node.ts

```diff
@@ -231,8 +233,18 @@ export class Git implements INodeType {
 					}
 				}
 
+				const gitConfig: string[] = [];
+				const deploymentConfig = Container.get(DeploymentConfig);
+				const isCloud = deploymentConfig.type === 'cloud';
+				const securityConfig = Container.get(SecurityConfig);
+				const disableBareRepos = securityConfig.disableBareRepos;
+				if (isCloud || disableBareRepos) {
+					gitConfig.push('safe.bareRepository=explicit');
+				}
```

Comment:
since this change would be breaking, at least for cloud, please add entry to `packages/cli/BREAKING-CHANGES.md`